src/jit/compiler.cpp

```diff
@@ -4598,6 +4598,10 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
         codeGen->regSet.rsMaskResvd |= RBM_OPT_RSVD;
         assert(REG_OPT_RSVD != REG_FP);
     }
+    // compRsvdRegCheck() has read out the FramePointerUsed property, but doLinearScan()
+    // tries to overwrite it later. This violates the PhasedVar rule and triggers an assertion.
+    // TODO: What is the proper way to handle this situation?
+    codeGen->resetFramePointerUsedWritePhase();
```

Comment:
I would suggest changing the `TODO' into `TODO-ArmArch-Bug?` as this may or may not represent a bug.

---
In reply to: [93383976](https://github.com/dotnet/coreclr/pull/8680#discussion_r93383976) [](ancestors = 93383976)